src/main/java/org/jabref/gui/externalfiles/ImportHandler.java

```diff
@@ -109,6 +109,9 @@ protected List<ImportFilesResultItemViewModel> call() {
                 CompoundEdit ce = new CompoundEdit();
                 for (final Path file : files) {
                     final List<BibEntry> entriesToAdd = new ArrayList<>();
+                    Path parentPath = file.getParent();
+                    Path rootPath = parentPath.getParent();
+                    Path relPath = rootPath.relativize(file);
```

Comment:
This is wrong. Please learn about directories in JabRef at https://docs.jabref.org/finding-sorting-and-cleaning-entries/filelinks#directories-for-files.

The method you should use is `org.jabref.logic.externalfiles.LinkedFileHandler#relativize`.